public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -423,7 +423,12 @@ export class LokiDatasource
     return res.data ?? (res || []);
   }
 
-  async getQueryStats(query: string): Promise<QueryStats> {
+  async getQueryStats(query: string): Promise<QueryStats | undefined> {
+    // if query is invalid, clear stats, and don't request
+    if (!isValidQuery(query)) {
+      return undefined;
```

Comment:
This might be a nit, but I think there's a semantic difference between `null` and `undefined`. I think in this case `null` would be rather the value we should return, as we deliberately chose to return "defined nothing".